pkg/sdkserver/sdkserver.go

```diff
@@ -985,41 +999,238 @@ func (s *SDKServer) GetList(ctx context.Context, in *alpha.GetListRequest) (*alp
+		// not verify here.
+		for _, val := range list.Values {
+			if in.Value == val {
+				return nil, errors.Errorf("ALREADY_EXISTS. Value: %s already in List: %s", in.Value, in.Name)
+			}
+		}
+		batchList.listAppend = append(batchList.listAppend, in.Value)
+		s.gsListUpdates[in.Name] = batchList
+		// Queue up the Update for later batch processing by updateLists.
+		s.workerqueue.Enqueue(cache.ExplicitKey(updateLists))
+		return &alpha.List{}, nil
+	}
+	return nil, errors.Errorf("NOT_FOUND. %s List not found", in.Name)
 }
 
-// RemoveListValue returns the updated List.
+// RemoveListValue collapses all remove a value from a List requests into a single UpdateList request.
+// Returns NOT_FOUND if the List does not exist.
+// Returns NOT_FOUND if the value is not in the List.
 // [Stage:Alpha]
 // [FeatureFlag:CountsAndLists]
 func (s *SDKServer) RemoveListValue(ctx context.Context, in *alpha.RemoveListValueRequest) (*alpha.List, error) {
 	if !runtime.FeatureEnabled(runtime.FeatureCountsAndLists) {
 		return nil, errors.Errorf("%s not enabled", runtime.FeatureCountsAndLists)
 	}
-	// TODO(#2716): Implement Me
-	return nil, errors.Errorf("Unimplemented -- RemoveListValue coming soon")
+
+	s.logger.WithField("name", in.Name).Debug("Remove List Value")
+
+	s.gsUpdateMutex.Lock()
+	defer s.gsUpdateMutex.Unlock()
+
+	gs, err := s.gameServer()
+	if err != nil {
+		return nil, err
+	}
+
+	if list, ok := gs.Status.Lists[in.Name]; ok {
+		// Verify value exists in the list
+		for _, val := range list.Values {
+			if in.Value != val {
+				continue
+			}
+			// Add value to remove to gsListUpdates map.
+			batchList := s.gsListUpdates[in.Name]
+			batchList.listDelete = append(batchList.listDelete, in.Value)
+			s.gsListUpdates[in.Name] = batchList
+			// Queue up the Update for later batch processing by updateLists.
+			s.workerqueue.Enqueue(cache.ExplicitKey(updateLists))
+			return &alpha.List{}, nil
+		}
+		return nil, errors.Errorf("NOT_FOUND. Value: %s not found in List: %s", in.Value, in.Name)
+	}
+	return nil, errors.Errorf("NOT_FOUND. %s List not found", in.Name)
+}
+
+// updateList updates the Lists in the GameServer's Status with the batched update list requests.
+// Includes all SetCapacity, AddValue, and RemoveValue requests in the batched request.
+func (s *SDKServer) updateList(ctx context.Context) error {
+	gs, err := s.gameServer()
+	if err != nil {
+		return err
+	}
+	gsCopy := gs.DeepCopy()
+
+	s.logger.WithField("batchListUpdates", s.gsListUpdates).Debug("Batch updating List(s)")
+	s.gsUpdateMutex.Lock()
+	defer s.gsUpdateMutex.Unlock()
+
+	names := []string{}
+
+	for name, listReq := range s.gsListUpdates {
+		list, ok := gsCopy.Status.Lists[name]
+		if !ok {
+			continue
+		}
+		if listReq.capacitySet != nil {
+			list.Capacity = *listReq.capacitySet
+		}
+		if len(listReq.listDelete) != 0 {
+			// TODO: Is there a quicker way of doing this while still keeping the ordering in tact?
```

Comment:
Not that I can think of 🤔 The only potential thought I have that might be faster, is keep the list of deleted items in a Map, since the order of deleted item's shouldn't matter (I don't think?) - that would mean you could likely remove all values in a single loop, rather than a loop of loops.

Reply:
I mucked around with it a bit, and if we delete values of the list as we're going through it it will mess with the index, so that `list.Values = append(list.Values[:i], list.Values[i+1:]...)` would no longer delete the correct index,or the index may no longer exist. We could create a new slice and write the non-deleted values to that new slice, although I don't know if that will make much of a difference in terms of actual execution time?

Reply:
> We could create a new slice and write the non-deleted values to that new slice, although I don't know if that will make much of a difference in terms of actual execution time?

Yeah, that's how I'd probably do it - create a new slice, loop through the existing one, check the map and if the current value doesn't exist in the deletion map, add it to the new slice, and that new slice is the source of truth.

Uses more space (but not really a lot), but should be O(n) in time -- and I think it's a simpler implementation?